src/models/roomModel.js

```diff
@@ -0,0 +1,55 @@
+import { firestore} from '../firebase'
+
+  
+  export const getRoomDb = async (roomCode) => {
+    try {
+      const roomRef = firestore.collection('rooms').where('roomCode','==',roomCode)
+      const snapshot = await roomRef.get()
+      if (snapshot.docs.length === 0) {
+        return console.error(`No such room exists: ${roomCode}`)
+      }      
+      else {
+        const roomDoc = snapshot.docs[0]
+        const room = { ...roomDoc.data(), id: roomDoc.id }
+        return room
+      }
+    } catch (err) {
+      new Error(`Error at getRoomDb: ${err}`)
+    }
+  }
+  
+
+
+export const updateRoomParticipantsDb = async (id, userId) => {
+  
+    const roomRef = firestore.collection('rooms').doc(id)
+    roomRef.get().then((doc) => {
+      if (doc.exists) {
+        if (doc.data().numParticipants == null) {
+          roomRef.set({ // or should it be set
+            // numParticipants: 1,
+            participants: [userId]
```

Comment:
Could we change the `participants` here to `participantIds`?

Reply:
Done